### ISSUE TITLE
Allow users to choose between `LinkedClone` or `FullClone` mode for vSphere provider

### ISSUE THREAD

**Issue by abhinavmpandey08** [aws/eks-anywhere#3467]
**What would you like to be added**:
Add option in EKS Anywhere on vSphere to allow users to specify whether to use LinkedClone or FullClone mode for OS images.

**Why is this needed**:
Currently, EKS Anywhere on vSphere is hardcoded to use `LinkedClone` mode. This makes the VM provisioning times faster but doesn't allow users to configure the disk size on VMs.

Because of the way LinkedClone works, if the template has snapshots, then the VM size isn't configurable.
So, if the users want to modify the disk size, they have to delete the template snapshots for the VMs to fall back to FullClone.

**Comment by abhinavmpandey08** [comment]
more context on Full vs Linked clone mode: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/fae6ef88467e608665e2902e2bb0aaeb4cee67ed/apis/v1alpha4/types.go#L43-L52

**Comment by echel0n** [comment]
I agree that this is a important feature to be added in as I find the default 25GB disk space is not enough for some clusters and keeps causing disk pressure

**Comment by abhinavmpandey08** [comment]
This ticket should be resolved with this PR https://github.com/aws/eks-anywhere/pull/4634
These changes will be available in v0.15 release
